# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1325871589 18000
#      Fri Jan 06 12:39:49 2012 -0500
# Branch stable
# Node ID ebcb0f83698f976ef91de40e1cc09cb1c29bf451
# Parent  bb5790c5e713056b3da3241129a0543cd4a21ec5
use gnulib::isatty

* bootstrap.conf (gnulib_modules): Include isatty in the list.
* sysdep.cc (raw_mode): Use gnulib::isatty.
* octave.cc (octave_main): Likewise.
* lex.ll: Define isatty to use gnulib namespace.

diff --git a/build-aux/bootstrap.conf b/build-aux/bootstrap.conf
--- a/build-aux/bootstrap.conf
+++ b/build-aux/bootstrap.conf
@@ -31,16 +31,17 @@ gnulib_modules="
   fflush
   fseek
   ftell
   getcwd
   gethostname
   getopt-gnu
   gettimeofday
   glob
+  isatty
   link
   lstat
   mkdir
   mkfifo
   mkostemp
   mkstemp
   mktime
   nanosleep
diff --git a/src/octave.cc b/src/octave.cc
--- a/src/octave.cc
+++ b/src/octave.cc
@@ -829,19 +829,20 @@ octave_main (int argc, char **argv, int 
 
   atexit (do_octave_atexit);
 
   // Is input coming from a terminal?  If so, we are probably
   // interactive.
 
   // If stdin is not a tty, then we are reading commands from a pipe or
   // a redirected file.
-  stdin_is_tty = isatty (fileno (stdin));
+  stdin_is_tty = gnulib::isatty (fileno (stdin));
 
-  interactive = (! embedded && stdin_is_tty && isatty (fileno (stdout)));
+  interactive = (! embedded && stdin_is_tty
+                 && gnulib::isatty (fileno (stdout)));
 
   if (! interactive && ! forced_line_editing)
     line_editing = false;
 
   // Force default line editor if we don't want readline editing.
   if (! line_editing)
     command_editor::force_default_editor ();
 
diff --git a/src/sysdep.cc b/src/sysdep.cc
--- a/src/sysdep.cc
+++ b/src/sysdep.cc
@@ -283,17 +283,17 @@ sysdep_cleanup (void)
 // It doesn't matter whether an input \n is mapped to \r, or vice versa.
 
 void
 raw_mode (bool on, bool wait)
 {
   static bool curr_on = false;
 
   int tty_fd = STDIN_FILENO;
-  if (! isatty (tty_fd))
+  if (! gnulib::isatty (tty_fd))
     {
       if (interactive)
         error ("stdin is not a tty!");
       return;
     }
 
   if (on == curr_on)
     return;
